game/tests/test_services.py

```
@@ -335,3 +335,84 @@ def test_delete_active_game(self):
+
+    # Game 2 state must reflect its own mode, not be corrupted by game 1
+    state_response_2 = self.client.get(f'/api/state/?game_id={game_id_2}')
+    self.assertEqual(state_response_2.status_code, 200)
+    self.assertEqual(state_response_2.json()['mode'], 'ai')
+
+
+def test_api_move_requires_game_id(self):
+    """POST /api/move/ without game_id must return 400, not process blindly."""
+    response = self.client.post(
+        '/api/move/',
+        data=json.dumps({'from_row': 6, 'from_col': 4, 'to_row': 4, 'to_col': 4}),
```

Ruff (0.15.20)

[error] Undefined name `json`

(F821)